app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -287,6 +287,7 @@ def sync_complete
     def remove_content
       fail _("No content ids provided") if @content.blank?
       respond_for_async :resource => sync_task(::Actions::Katello::Repository::RemoveContent, @repository, @content)
+      async_task(::Actions::Katello::Repository::CapsuleGenerateAndSync, @repository)
```

Comment:
From inside the task, as in the RemoveContent task's finalize phase should call this instead of the controller. @jlsherrill would you concur?